### ISSUE TITLE
Question: for unit testing an object, use scenarios+feature file like functional tests?

### ISSUE THREAD

**Issue by ghost** [olegpidsadnyi/pytest-bdd-example#6]
For unit testing an object/class, what is the right practice? Do I use scenario/feature/given/when/then functionnalities like I use for functional tests? (eg. If I want to a unit test Calculator() object do I create a feature/test_calculator.feature ?) or is it overkill for unit testing?

Or I just use "standard" pytest testing without the scenario/feature/given/when/then functionnalities?

I read "BDD in Action" and they show 3 style practices for unit tests:

Practice 1) (book page 271) equivalent to [fluent-test](https://fluent-test.readthedocs.io/en/latest/)?

```
class WhenX_Y_Z()
def should_do_something()
      expect(balbalblabla)
```

Practice 2) (book page 282)

```
class WhenX_Y_Z()
def should_do_something()
     // implicit GIVEN
     a = a()
     // implicit WHEN
      a.something()
     //implicit THEN
      expect(a.balbalblabla)
```

Practice 3) (book page 288) equivalent to [pytest-describe](https://github.com/ropez/pytest-describe) and [mamba](https://github.com/nestorsalceda/mamba)? 

```
describe(........)
      it(..........................)
```

---

Practice 4) [from here](https://www.paragon-inc.com/resources/blogs-posts/givenwhenthen-writing-unit-tests-in-3-lines)

```

def should_x_yz_():
     given(.....)
     when(.......)
      then(......)

def given(........):
              .......

def when(...............................):
  .........................

def then(......):
                           .....................

```

I'm quite lost in how the best practice to implement unit test in pytest-bdd 
So basically if Oleg or someone could show me example of unit test with pytest-bdd I would be happy

The only thing I know is that I don't want a separate test runner so  [mamba](https://github.com/nestorsalceda/mamba) would not qualify as it is not integrated with pytest

**Comment by ghost** [comment]
for the record I went with pytest-describe
